File: hackertracker/src/main/java/com/shortstack/hackertracker/Renderer/InformationRenderer.java
package com.shortstack.hackertracker.Renderer;

import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.pedrogomez.renderers.Renderer;
import com.shortstack.hackertracker.BottomSheet.InformationBottomSheetDialogFragment;
import com.shortstack.hackertracker.Model.Information;
import com.shortstack.hackertracker.R;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class InformationRenderer extends Renderer<Information> implements View.OnClickListener {

    @BindView(R.id.header)
    TextView header;

    @Override
    protected View inflate(LayoutInflater inflater, ViewGroup parent) {
        View view = inflater.inflate(R.layout.row_info, parent, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    protected void hookListeners(View rootView) {
        rootView.setOnClickListener(this);
    }

    @Override
    public void render(List<Object> payloads) {
        header.setText(getContent().getTitle());
    }


    @Override
    public void onClick(View view) {
        InformationBottomSheetDialogFragment badges = InformationBottomSheetDialogFragment.newInstance(getContent());
        badges.show(((AppCompatActivity)getContext()).getSupportFragmentManager(), badges.getTag());
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/ui/information/renderers/InformationRenderer.kt
package com.shortstack.hackertracker.ui.information.renderers

import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.pedrogomez.renderers.Renderer
import com.shortstack.hackertracker.R
import com.shortstack.hackertracker.models.Information
import com.shortstack.hackertracker.ui.information.InformationBottomSheet
import kotlinx.android.synthetic.main.row_info.view.*

class InformationRenderer : Renderer<Information>(), View.OnClickListener {

    override fun inflate(inflater: LayoutInflater, parent: ViewGroup): View {
        return inflater.inflate(R.layout.row_info, parent, false)
    }

    override fun hookListeners(rootView: View?) {
        rootView?.setOnClickListener(this)
    }

    override fun render(payloads: List<Any>) {
        rootView.header.text = content.title
    }


    override fun onClick(view: View) {
        val badges = InformationBottomSheet.newInstance(content)
        badges.show((context as AppCompatActivity).supportFragmentManager, badges.tag)
    }
}
